src/Main/MainGui.cpp

```diff
@@ -95,6 +95,35 @@ class Redirection
     FILE* file;
 };
 
+static void DisplayInfo(const QString& msg, bool preformatted = true)
+{
+    if (App::Application::Config()["Console"] == "1") {
+        std::cout << msg.toLatin1().constData();
+        return;
+    }
+
+    QString appName = QString::fromLatin1(App::Application::Config()["ExeName"].c_str());
+    QMessageBox msgBox;
+    msgBox.setIcon(QMessageBox::Information);
+    msgBox.setWindowTitle(appName);
+    msgBox.setDetailedText(msg);
+    msgBox.setText(preformatted ? QLatin1String("<pre>") + msg + QLatin1String("</pre>") : msg);
+    msgBox.exec();
+}
+
+static void DisplayCritical(const QString& msg, bool preformatted = true)
+{
+    if (App::Application::Config()["Console"] == "1") {
+        std::cerr << msg.toLatin1().constData();
+        return;
+    }
+
+    QString appName = QString::fromLatin1(App::Application::Config()["ExeName"].c_str());
+    QString title = QObject::tr("Initialization of %1 failed").arg(appName);
+    QString text = preformatted ? QLatin1String("<pre>") + msg + QLatin1String("</pre>") : msg;

```

Comment:
How about using `.arg()` here too?
```c++
QString text = preformatted ? QStringLiteral("<pre>%1</pre>").arg(msg) : msg;
```

Reply:
`QLatin1String` has the `arg` method only added in Qt-5.14; on Ubuntu  20.04 we still have 5.12, though.
Next try with the `QStringLiteral` macro...
(I should have read your comment more carefully in the first place)